src/main/java/pl/project13/core/GitCommitIdPlugin.java

```diff
@@ -367,6 +383,9 @@ private static void loadGitDataWithNativeGit(
       @Nonnull Callback cb,
       @Nonnull File dotGitDirectory,
       @Nonnull Properties properties) throws GitCommitIdExecutionException {
+    if (cb.getPerModuleVersions()) {
+      throw new GitCommitIdExecutionException("The native git provider does not support per module versions.");
```

Comment:
First thanks for your contribution, maybe it is due to the fact that I never fully understood this "per module" request, but could you outline why the native git does not support this feature (maybe by explaining what this "per module" even means?).
I personally would kinda dislike the fact that certain features of the plugin are not supported by all providers (jgit/git)....